### PR TITLE
Check negative inclusion fee

### PR DESCRIPTION
**Pull request by sisuresh**
# Description

# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: src/herder/TransactionQueue.cpp [hunk 1]
```diff
@@ -116,10 +116,10 @@ canReplaceByFee(TransactionFrameBasePtr tx, TransactionFrameBasePtr oldTx,
     //
     // FEE_MULTIPLIER * v2 does not overflow uint128_t because fees are bounded
     // by INT64_MAX, while number of operations and FEE_MULTIPLIER are small.
-    uint128_t v1 = bigMultiply(newFee, oldNumOps);
     uint128_t v2 = bigMultiply(oldFee, newNumOps);
     uint128_t minFeeN = v2 * TransactionQueue::FEE_MULTIPLIER;
-    bool res = v1 >= minFeeN;
+
+    bool res = newFee >= 0 && bigMultiply(newFee, oldNumOps) >= minFeeN;
```
Comment (dmkozh) [inline src/herder/TransactionQueue.cpp:122]: This check is good and makes sense to have here. But I would also add a simple negative inclusion fee check before adding tx to the limiter in the first place (IIUC that's `canAdd` method in tx queue).

**Review by dmkozh — APPROVED** [review]

File: src/herder/TransactionQueue.cpp [hunk 2]
```diff
@@ -116,10 +116,10 @@ canReplaceByFee(TransactionFrameBasePtr tx, TransactionFrameBasePtr oldTx,
     //
     // FEE_MULTIPLIER * v2 does not overflow uint128_t because fees are bounded
     // by INT64_MAX, while number of operations and FEE_MULTIPLIER are small.
-    uint128_t v1 = bigMultiply(newFee, oldNumOps);
     uint128_t v2 = bigMultiply(oldFee, newNumOps);
```
[the remark this answers is not in the document]
Comment (sisuresh) [reply src/herder/TransactionQueue.cpp]: Updated to `oldTotalFee`.

File: src/herder/TransactionQueue.cpp (hunk 1 above)
Comment (sisuresh) [reply src/herder/TransactionQueue.cpp:123]: Added a check close to the top of `canAdd`.

**Comment by dmkozh** [comment]
r+ 6a6f78e

**Pull request merged**